assets/source/setup-guide/app/data/settings/reducer.js

```diff
@@ -42,6 +42,14 @@ const settingsReducer = (
 				isUpdating: false,
 			};
 			break;
+		case TYPES.SET_UPDATED_DATA:
+			state = {
+				...state,
+				updatedData: action.reset
+					? {}
+					: { ...state.updatedData, ...action.data },
+			};
+			break;
```

Comment:
💅  I noticed that there is no default value for `updatedData` in the settings reducer, and it brought the following behaviour (very minor):

- Go to `/wp-admin/admin.php?page=wc-admin&path=/pinterest/settings`
- Click `Save changes` straightaway
- From the dev console the `updatedData` is `undefined`
- Click `Save changes` again
- From the dev console the `updatedData` is an empty object `{}`

When the data/analysis people try to filter out how many clicks that has no data change, the filter condition becomes inconsistent.

Reply:
Amazing catch! 🕵️ 

Fixed in [429af73](https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/429af73ca403453f802fed7d34e43d44b784dac4)